src/services/auth.service.ts

```diff
@@ -0,0 +1,71 @@
+import * as firebase from "firebase-admin";
+import * as uno from "uno-serverless";
+import { memoize } from "uno-serverless";
+
+export type UserRecord = firebase.auth.UserRecord;
+
+export interface AuthService {
+  /**
+   * Initialize the service with the value of the HTTP Authorization header.
+   */
+  init(authorizationHeader: string): Promise<void>;
+
+  /**
+   * Return whether or not a user is authenticated.
+   */
+  isAuthenticated(): boolean;
+
+  /**
+   * Returns a promise that  which will resolve with the authenticated user.
+   *
+   * If no user is authenticated the promise will reject.
+   */
+   authenticatedUser(): Promise<UserRecord>;
```

Comment:
Not sure this belongs here. We could restraints the authentication service and instead have a users service which. To get the information about the authentication user consumers would first get their UID using authenticatedUserId and use that to retrieve the user record from the users service.

Reply:
je crois que tu as raison, mais pour l'instant, ça serait good enough, on pourrait ouvrir un issue pour avoir un user service, qu'est-ce que tu en penses?

Reply:
Yep, on peut faire sans user service pour le moment et changer ca plus tard.
En fait `AuthService.authenticatedUser` devrait simplement déleguer au user service une fois qu'il existe.